src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1181,7 +1170,7 @@ EnsureReplicationSettings(Oid relationId, char replicationModel)
 						errmsg("replication factors above one are incompatible with %s",
 							   msgSuffix),
 						errhint("Try again after reducing \"citus.shard_replication_"
-								"factor\" to one%s.", extraHint)));
+								"factor\" to one.")));
```

Comment:
is this still accurate / useful?

Reply:
I should add more logic and improve the error messages here. Consider this scenario:

```sql
set citus.shard_replication_factor = 2;
SELECT create_distributed_table('table1', 'id');

set citus.shard_replication_factor = 1;
SELECT create_distributed_table('table2', 'id', colocate_with => 'table1');
```

When colocation parameter is given, Citus should prefer the replication model used by the colocated table, and ignore the one decided by `citus.shard_replication_factor`.

I believe I fixed the issue above, but we still see this error message in the test outputs in some places. I believe I should remove/change the error message here after making sure I fix all the occurrences in test files such as the following:

https://github.com/citusdata/citus/blob/deb07169d76a28ccdf531f1af81db792883d3a90/src/test/regress/expected/alter_distributed_table.out#L678-L681